client/components/platform-checkout/hooks/use-platform-checkout-user.js

```diff
@@ -1,6 +1,31 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from 'react';
+
 // hook for handling API calls to get and create platform checkout user.
 const usePlatformCheckoutUser = () => {
-	const isRegisteredUser = true;
+	const [ isRegisteredUser, setIsRegisteredUser ] = useState( false );
+	const windowRef = useRef( window );
+
+	useEffect( () => {
+		const handlePlatformCheckoutUserCheck = ( e ) => {
+			setIsRegisteredUser( e.detail.isRegisteredUser );
+		};
+
+		const currentWindowRef = windowRef.current;
+		currentWindowRef.addEventListener(
+			'PlatformCheckoutUserCheck',
+			handlePlatformCheckoutUserCheck
+		);
+
+		return () => {
+			currentWindowRef.removeEventListener(
+				'PlatformCheckoutUserCheck',
+				handlePlatformCheckoutUserCheck
+			);
+		};
+	}, [] );
 
 	return {
```

Comment:
What are your thoughts on also just returning the boolean value here, rather than an object containing the boolean?
The boolean can be uglified by webpack, whereas an object and its properties can't

Reply:
That's a good point. I sat here for a while trying to think of a use case where we would need to add something to the return value in the future and came up empty. We can always change what's returned in the future anyway. I'll make that update to the hook then update the usage and tests to expect an unnamed boolean. 👍

Reply:
Resolved in b1f7ace3